srml/contract/src/wasm/prepare.rs

```diff
@@ -36,9 +36,12 @@ struct ContractModule<'a, Gas: 'a> {
 }
 
 impl<'a, Gas: 'a + As<u32> + Clone> ContractModule<'a, Gas> {
-	fn new(original_code: &[u8], schedule: &'a Schedule<Gas>) -> Result<ContractModule<'a, Gas>, Error> {
+	fn new(
+		original_code: &[u8],
+		schedule: &'a Schedule<Gas>,
+	) -> Result<ContractModule<'a, Gas>, &'static str> {
```

Comment:
I feel like the move from an explicit error type to generic `'static str` is a regression. What is the reason behind that?

Reply:
Hm, and you're not alone!

https://github.com/paritytech/substrate/pull/1358#discussion_r246047220

Why exactly you think it is a regression?

Reply:
Just wrote a bunch of tests (see `prepare_test!`) and looking at the result convienced me even more about superiority of strings in this particular case!

Resolving this issue, but feel free to unresolve if you don't agree!